impl/affinity.cxx

```diff
@@ -131,13 +136,26 @@ affinity_t::affinity_t(std::size_t threadCount, const std::vector<uint32_t>& pin
 			}
 		}
 	}
+
+	if (processors.empty())
+		throw std::runtime_error("At least one core should be available and pinned");
 #else
 	(void)threadCount;
 	(void)pinning;
 	throw std::runtime_error("Not implemented");
 #endif
 }
 
+affinity_t::affinity_t(std::size_t threadCount, std::initializer_list<uint32_t> pinning)
+	: affinity_t(threadCount, std::vector<uint32_t>{pinning})
```

Comment:
The `:` goes on the end of the previous line please, with the empty body on the end of this line in the form `{ }`